tools/eval/datasets/widerface.py

```diff
@@ -0,0 +1,309 @@
+            event_name = str(event_list[i][0][0])
+            img_list = file_list[i][0]
+            pred_list = pred[event_name]
+            sub_gt_list = gt_list[i][0]
+            # img_pr_info_list = np.zeros((len(img_list), thresh_num, 2))
+            gt_bbx_list = facebox_list[i][0]
+
+            for j in range(len(img_list)):
+                pred_info = pred_list[str(img_list[j][0][0])]
+
+                gt_boxes = gt_bbx_list[j][0].astype('float')
+                keep_index = sub_gt_list[j][0]
+                count_face += len(keep_index)
+
+                if len(gt_boxes) == 0 or len(pred_info) == 0:
+                    continue
+                ignore = np.zeros(gt_boxes.shape[0])
+                if len(keep_index) != 0:
+                    ignore[keep_index - 1] = 1
+                pred_recall, proposal_list = image_eval(pred_info, gt_boxes, ignore, iou_thresh)
+
+                _img_pr_info = img_pr_info(thresh_num, pred_info, proposal_list, pred_recall)
+
+                pr_curve += _img_pr_info
+        pr_curve = dataset_pr_info(thresh_num, pr_curve, count_face)
+
+        propose = pr_curve[:, 0]
+        recall = pr_curve[:, 1]
+
+        ap = voc_ap(recall, propose)
+        aps.append(ap)
+    return aps
+
+
+class WIDERFace:
+    def __init__(self, root, split='val'):
+        self.aps = []
+        self.widerface_root = root
+        self._split = split
+
+        self.widerface_img_paths = {
+            'val': os.path.join(self.widerface_root, 'WIDER_val', 'images'),
+            'test': os.path.join(self.widerface_root, 'WIDER_test', 'images')
+        }
+
+        self.widerface_split_fpaths = {
+            'val': os.path.join(self.widerface_root, 'wider_face_split', 'wider_face_val.mat'),
+            'test': os.path.join(self.widerface_root, 'wider_face_split', 'wider_face_test.mat')
+        }
+        self.img_list, self.num_img = self.load_list()
+
+    def load_list(self):
+        n_imgs = 0
+        flist = []
+
+        split_fpath = self.widerface_split_fpaths[self._split]
+        img_path = self.widerface_img_paths[self._split]
+
+        anno_data = loadmat(split_fpath)
+        event_list = anno_data.get('event_list')
+        file_list = anno_data.get('file_list')
+
+        for event_idx, event in enumerate(event_list):
+            event_name = event[0][0]
+            for f_idx, f in enumerate(file_list[event_idx][0]):
+                f_name = f[0][0]
+                f_path = os.path.join(img_path, event_name, f_name + '.jpg')
+                flist.append(f_path)
+                n_imgs += 1
+
+        return flist, n_imgs
+
+    def __getitem__(self, index):
+        img = cv.imread(self.img_list[index])
+        event, name = self.img_list[index].split(os.sep)[-2:]
+        return event, name, img
+
+    def eval(self, model):
+        results_list = dict()
+        # forward
+        for event_name, img_name, img in tqdm.tqdm(self):
```

Comment:
Can you provide a screenshot on the output through the whole process? I suggest to print the model name as well.

Reply:
I forgot to output the model name, I will add it in the next update.